### ISSUE TITLE
Roslyn should not infer oblivious annotations when nullable context is enabled

### ISSUE THREAD

**Issue by TessenR** [dotnet/roslyn#36541]
**Version Used**: 

Branch master (18 Jun 2019)
Latest commit d4afae by Andrew Hall:
Disable move type when the options service isn't present (#36334)

Use MEF to provide implementation contract

**Steps to Reproduce**:
Compile the following code:
```cs
#nullable enable
interface I1<T> { T Prop { get; set; } }

class D
{
    void M(ref I1<I1<string?>> i1, ref I1<I1<string>?> i2, bool b)
    {
        var x = b ? ref i1 : ref i2;

        x.Prop.Prop.ToString();
        x.Prop.Prop = null;
        x.Prop.ToString();
        x.Prop = null;
    }        
}
```

**Expected Behavior**:
`x` is inferred to some type you can write explicitly. Consequently `x.Prop` and `x.Prop.Prop` are either nullable or non-nullable and either dereferneces of these properties or assignments of null values to them reported as warnings.

**Actual Behavior**:
No warnings are reported for usages of `x` and its properties.

**Notes**
It looks like `x` is inferred to a type with oblivious annotations (except for the top-level annotation which is notnull) since neither dereferences of `x.Prop` and `x.Prop.Prop` nor assigning `null` values to them produce any warnings.

Note that you cannot achieve the same warnings behavior with explicitly specified type usage. I.e.
`I1<I1<string>> x = ...` produces warnings for assignments of null values
`I1<I1<string?>> x = ...` produces warnings for the first dereference of `x.Prop.Prop` and assignment of null to `x.Prop`
`I1<I1<string>?> x = ...` produces warnings for the first dereference of `x.Prop` and assignment of null to `x.Prop.Prop`
`I1<I1<string?>?> x = ...` produces warnings for the first dereferences of `x.Prop` and `x.Prop.Prop`

**Comment by jcouv** [comment]
Strangely, this scenario currently produces a warning on the conditional ([sharplab](https://sharplab.io/#v2:EYLgtghgzgLgpgJwDQxASwDYB8DEA7AVwwwmAzgAI49TyBYAKDT3gQDMIBjSgSQEYAPABUAfBQDeFIRQAKCAPYAHCRQDmcGAG4KUDdoC+FfY0YABAEwUAIo3GMKDiqYAsFALIAKBHDYV+A/1M+AAYAfhExND4kCm9ff0CQkXCKNHMY4Hl5DApgAEp7RzsGR1KKADcIBAoADwoAXlyKUNifVL4KEFbfNO0Aej6KAHcqvGZVCgBhAGUADgA2PgBOLoA5IhJgTDQYAE8KeV84xGpuCj3FOChUvAqIDAJKQ/Pdy4oAcgTBILCI94oACbyK54d4wCiQGCcAAW5yq6nBF0on0EiWCyRE7wAdCYSmUHDUsXIlESFIosUJ5NMYAhxh48ppCvjCcTyayGhRCMRGXjmaSSZTqbS8Kp6Tz8QT+cpGlyMOLHIYJYx9EA)):

```C#
#nullable enable
interface I1<T> { T Prop { get; set; } }

class D
{
    void M(ref I1<I1<string?>> i1, ref I1<I1<string>?> i2, bool b)
    {
        var x = b ? ref i1 : ref i2; // warning CS8619: Nullability of reference types in value of type 'I1<I1<string?>>' doesn't match target type 'I1<I1<string>?>'.
...
    }        
}```
